x-pack/plugins/features/server/feature_registry.ts

```diff
@@ -169,19 +190,106 @@ export class FeatureRegistry {
     }
   }
 
-  public getAllKibanaFeatures(license?: ILicense, ignoreLicense = false): KibanaFeature[] {
+  /**
+   * Once all features are registered and the registry is locked, this method should validate the integrity of the registered feature set, including any potential cross-feature dependencies.
+   */
+  public validateFeatures() {
```

Comment:
__note:__ I don't perform any validation during feature registration because it's only possible to do a limited number of checks, as not all features are registered at the `setup` stage and I didn't want to spread the validation logic across multiple lifecycle methods.